store/store.go

```diff
@@ -498,49 +457,44 @@ func (s *Store[H]) flush(ctx context.Context, headers ...H) error {
 	return batch.Commit(ctx)
 }
 
-// loadContiguousHead from the disk and sets contiguousHead and heightSub.
-func (s *Store[H]) loadContiguousHead(ctx context.Context) error {
-	h, err := s.readHead(ctx)
-	if err != nil {
-		return err
-	}
-
-	s.contiguousHead.Store(&h)
-	s.heightSub.SetHeight(h.Height())
-	return nil
-}
-
-// readHead loads the head from the datastore.
-func (s *Store[H]) readHead(ctx context.Context) (H, error) {
+// readByKey the hash under the given key from datastore and fetch the header by hash.
+func (s *Store[H]) readByKey(ctx context.Context, key datastore.Key) (H, error) {
 	var zero H
-	b, err := s.ds.Get(ctx, headKey)
+	b, err := s.ds.Get(ctx, key)
 	if err != nil {
 		return zero, err
 	}
 
-	var head header.Hash
-	err = head.UnmarshalJSON(b)
-	if err != nil {
+	var h header.Hash
+	if err := h.UnmarshalJSON(b); err != nil {
 		return zero, err
 	}
-
-	return s.Get(ctx, head)
+	return s.Get(ctx, h)
 }
 
-// readTail loads the tail from the datastore.
-func (s *Store[H]) readTail(ctx context.Context) (H, error) {
-	var zero H
-	b, err := s.ds.Get(ctx, tailKey)
+func (s *Store[H]) setTailHeader(h H) error {
+	b, err := h.Hash().MarshalJSON()
 	if err != nil {
-		return zero, err
+		return err
 	}
 
-	var tail header.Hash
-	if err := tail.UnmarshalJSON(b); err != nil {
-		return zero, err
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
```

Comment:
why not pass instead?

Reply:
Nothing to pass, we don't have a `ctx` where it's called (ok, technically there is `ctx` in `flushLoop` but that's a fake context).

Reply:
I think we can make it real as part of https://github.com/celestiaorg/go-header/pull/214